### PR TITLE
HW12 lfu-cache

### PR DESCRIPTION
**Pull request by b-bodnar**

### CODE REVIEW

File: src/main/java/cursor/hw12/model/Node.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@Getter
+@Setter
+public class Node {
+
+    private int averageSalary;
+    private int frequency;
+
+    public Node(int data) {
+        this.averageSalary = data;
+        this.frequency = 0;
```
Comment (YStakhniak) [inline src/main/java/cursor/hw12/model/Node.java]: Why you hard code "0" in the constructor?

File: src/main/java/cursor/hw12/service/LFUCacheService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package cursor.hw12.service;
+
+import cursor.hw12.dao.AverageSalaryDao;
+import cursor.hw12.model.Node;
+import java.util.Map;
+
+public class LFUCacheService {
+    private static final int capacity = 10;
```
Comment (YStakhniak) [inline src/main/java/cursor/hw12/service/LFUCacheService.java]: if it's a constant it should be private static final int CAPACITY = 10;

File: src/main/java/cursor/hw12/service/LFUCacheService.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private static Map<String, Node> cache;
+    public static LFUCacheService lfuCacheService;
+
+    private LFUCacheService() {
+        cache = AverageSalaryDao.getCache();
+    }
+
+    public static LFUCacheService getInstance() {
+        return lfuCacheService != null
```
Comment (YStakhniak) [inline src/main/java/cursor/hw12/service/LFUCacheService.java]: It's better to use method nonNull() from Objects class 
(Objects.nonNull(lfuCacheService))

File: src/main/java/cursor/hw12/service/LFUCacheService.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public Node getNode(String key) {
+        if (cache.containsKey(key)) {
+            Node node = cache.get(key);
+            node.setFrequency(node.getFrequency() + 1);
+            cache.put(key, node);
+            return node;
+        }
+        return null;
```
Comment (YStakhniak) [inline src/main/java/cursor/hw12/service/LFUCacheService.java]: Never return null ! Use Optional or other approach.

File: src/main/java/cursor/hw12/model/Node.java (hunk 1 above)
Comment (b-bodnar) [reply src/main/java/cursor/hw12/model/Node.java]: Resolved

File: src/main/java/cursor/hw12/service/LFUCacheService.java (hunk 2 above)
Comment (b-bodnar) [reply src/main/java/cursor/hw12/service/LFUCacheService.java]: Resolved

File: src/main/java/cursor/hw12/service/LFUCacheService.java (hunk 3 above)
Comment (b-bodnar) [reply src/main/java/cursor/hw12/service/LFUCacheService.java]: Resolved

File: src/main/java/cursor/hw12/service/LFUCacheService.java (hunk 4 above)
Comment (b-bodnar) [reply src/main/java/cursor/hw12/service/LFUCacheService.java]: Resolved

**Pull request merged**